dipy/tracking/local/tests/test_tissue_classifier.py

```diff
@@ -0,0 +1,173 @@
+import numpy as np
+import numpy.testing as npt
+import scipy.ndimage
+
+from dipy.tracking.local import (BinaryTissueClassifier,
+                                 ThresholdTissueClassifier,
+                                 ActTissueClassifier)
+
+
+def test_binary_tissue_classifier():
+    """This tests that the binary tissue classifier return expected
+     tissue types:
+
+        OUTSIDEIMAGE = -1
+        INVALIDPOINT = 0
+        TRACKPOINT = 1
+        ENDPOINT = 2
+    """
+    # enum TissueClass (tissue_classifier.pxd) is not accessible
+    # from here. To be changed when minimal cyhton version > 0.21.
+    # cython 0.21 - cpdef enum to export values into Python-level namespace
+    # https://github.com/cython/cython/commit/50133b5a91eea348eddaaad22a606a7fa1c7c457
+
+    mask = np.random.random((4, 4, 4))
+    mask[mask < 0.4] = 0.0
+
+    btc_boolean = BinaryTissueClassifier(mask > 0)
+    btc_float64 = BinaryTissueClassifier(mask)
+
+    # test voxel center
+    for ind in np.ndindex(mask.shape):
+        pts = np.array(ind, dtype='float64')
+        state_boolean = btc_boolean.check_point(pts)
+        state_float64 = btc_float64.check_point(pts)
+        if mask[ind] > 0:
+            npt.assert_equal(state_boolean, 1)
+            npt.assert_equal(state_float64, 1)
+        else:
+            npt.assert_equal(state_boolean, 2)
+            npt.assert_equal(state_float64, 2)
+
+    # test random points in voxel
+    for ind in np.ndindex(mask.shape):
+        for _ in range(10):
+            pts = np.array(ind, dtype='float64') + np.random.random(3) - 0.5
+            state_boolean = btc_boolean.check_point(pts)
+            state_float64 = btc_float64.check_point(pts)
+            if mask[ind] > 0:
+                npt.assert_equal(state_boolean, 1)
+                npt.assert_equal(state_float64, 1)
+            else:
+                npt.assert_equal(state_boolean, 2)
+                npt.assert_equal(state_float64, 2)
+
+    # test outside points
+    outside_pts = [[100, 100, 100], [0, -1, 1], [0, 10, 2]]
+    for pts in outside_pts:
+        pts = np.array(pts, dtype='float64')
+        state_boolean = btc_boolean.check_point(pts)
+        state_float64 = btc_float64.check_point(pts)
+        npt.assert_equal(state_boolean, -1)
+        npt.assert_equal(state_float64, -1)
```

Comment:
I would like to know also how this is doing with points that are really close to the boundary. E.g. (0.1, 0.1, 0.1), (0.5, 0.5, 0.5) and (-0.5, 0., 0).